Source/Public/Build-Bicep.ps1

```diff
@@ -0,0 +1,91 @@
+    Build-Bicep is equivalent to bicep build but with the possibility to compile all .bicep files in a directory.
+.PARAMETER Path
+    Specfies the path to the directory or file that should be compiled
+.PARAMETER ExcludeFile
+    Specifies a .bicep file to exclude from compilation
+.PARAMETER GenerateParameterFile
+    The -GenerateParameterFile switch generates a ARM Template paramter file for the compiled template
+.EXAMPLE
+    Build-Bicep vnet.bicep
+    Compile single bicep file in working directory
+.EXAMPLE
+    Build-Bicep 'c:\bicep\modules\vnet.bicep'
+    Compile single bicep file in different directory
+.EXAMPLE
+    Build-Bicep
+    Compile all .bicep files in working directory
+.EXAMPLE
+    Build-Bicep -Path 'c:\bicep\modules\'
+    Compile all .bicep files in different directory
+.EXAMPLE
+    Build-Bicep -ExcludeFile vnet.bicep
+    Compile all .bicep files in the working directory except vnet.bicep
+.EXAMPLE
+    Build-Bicep -Path 'c:\bicep\modules\' -ExcludeFile vnet.bicep
+    Compile all .bicep files in different directory except vnet.bicep
+.EXAMPLE
+    Build-Bicep -GenerateParameterFile
+    Compile all .bicep files in the working directory and generates a parameter files
+.NOTES
+    Go to module repository https://github.com/StefanIvemo/BicepPowerShell for detailed info, reporting issues and to submit contributions.
+#>
+function Build-Bicep {
+    [CmdletBinding(DefaultParameterSetName = 'Default')]
+    [Alias('Invoke-BicepBuild')]
+    param (
+        [Parameter(ParameterSetName = 'Default',Position=1)]
+        [Parameter(ParameterSetName = 'AsString',Position=1)]
+        [string]$Path = $pwd.path,
+
+        [Parameter(ParameterSetName = 'Default',Position=2)]
+        [Parameter(ParameterSetName = 'AsString',Position=2)]
+        [ValidateNotNullOrEmpty()]
+        [string]$OutputDirectory = $pwd.Path,
+
+        [Parameter(ParameterSetName = 'Default')]
+        [Parameter(ParameterSetName = 'AsString')]
+        [string[]]$ExcludeFile,
+
+        [Parameter(ParameterSetName = 'Default')]
+        [switch]$GenerateParameterFile,
+
+        [Parameter(ParameterSetName = 'AsString')]
+        [switch]$AsString
+    )
+
+    begin {
+        if (-not (Test-Path $OutputDirectory)) {
+            New-Item $OutputDirectory -Force -ItemType Directory
+        }
+    }
+
+    process {
+        $files = Get-Childitem -Path $Path *.bicep -File
+        if ($files) {
+            foreach ($file in $files) {
+                if ($file.Name -notin $ExcludeFile) {
+                    $ARMTemplate = ParseBicep -Path $file.FullName
+                    if ($AsString.IsPresent) {
+                        Write-Output $ARMTemplate
+                    }
+                    else {                        
+                        $OutputFilePath = Join-Path -Path $OutputDirectory -ChildPath ('{0}.json' -f $file.BaseName)
+                        $null = Out-File -Path $OutputFilePath -InputObject $ARMTemplate -Encoding utf8
```

Comment:
If I run the cmdlet without the assemblies loaded (did not have them in the correct folder). It will generate a couple of errors and then create an empty ARM template JSON file in the working directory. Can we verify that the assemblies have successfully been loaded? Or will it not be an issue when installing the module properly using Install-Module?

Reply:
This should not be a problem, the module should not load at all without the right assemblies.